### PR TITLE
feature: adds optional flag to filter out sold-out products

### PR DESCRIPTION
**Pull request by willopez**
Impact: minor
Type: feature

## Feature
Adds optional flag to filter out sold out products in `catalogItems` query

## Testing
1. In the storefront, add a boolean param named `isSoldOut` to the `catalogItems` query and set it to either `true` or `false`
2. Visit tag page, if `isSoldOut=false`, confirm that sold out products are not returned.

### CODE REVIEW

File: imports/plugins/core/catalog/server/no-meteor/queries/catalogItems.js [hunk 1]
```diff
@@ -11,16 +11,23 @@ import ReactionError from "@reactioncommerce/reaction-error";
...
   const { collections } = context;
   const { Catalog } = collections;
 
   if ((!shopIds || shopIds.length === 0) && (!tagIds || tagIds.length === 0)) {
     throw new ReactionError("invalid-param", "You must provide tagIds or shopIds or both");
   }
 
+  // If isSoldOut filter is provided, add it to the query
+  let isSoldOutFilter = {};
+  if (isSoldOut === true || isSoldOut === false) {
```
Comment (mikemurray) [inline imports/plugins/core/catalog/server/no-meteor/queries/catalogItems.js]: When testing for the type of something use `typeof isSoldOut === "boolean"`.

File: imports/plugins/core/catalog/server/no-meteor/queries/catalogItemsAggregate.js [hunk 2]
```diff
@@ -11,20 +11,27 @@ import ReactionError from "@reactioncommerce/reaction-error";
...
   const { collections } = context;
   const { Catalog, Tags } = collections;
 
   if ((!shopIds || shopIds.length === 0) && (!tagId)) {
     throw new ReactionError("invalid-param", "You must provide a tagId or shopIds or both");
   }
 
+  // If isSoldOut filter is provided, add it to the query
+  let isSoldOutFilter = {};
+  if (isSoldOut === true || isSoldOut === false) {
```
Comment (mikemurray) [inline imports/plugins/core/catalog/server/no-meteor/queries/catalogItemsAggregate.js]: `typeof isSoldOut === "boolean"`

**Comment by willopez** [comment]
@mikemurray comments resolved.

**Comment by mikemurray** [comment]
Tests are failing due to a security issue and should be fixed by this PR: https://github.com/reactioncommerce/reaction/pull/5108

**Pull request merged**